supervisor.go

```diff
@@ -0,0 +1,193 @@
+package druid
+
+import (
+	"strings"
+	"time"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	supervisorPathPrefix             = "druid/indexer/v1/supervisor"
+	supervisorSpecPathPrefix         = "druid/indexer/v1/supervisor/:supervisorId"
+	supervisorAllActivePathPrefix    = "druid/indexer/v1/supervisor?full"
+	supervisorStatusPathPrefix       = "druid/indexer/v1/supervisor/:supervisorId/status"
+	supervisorHistoryAllPathPrefix   = "druid/indexer/v1/supervisor/history"
+	supervisorHistoryPathPrefix      = "druid/indexer/v1/supervisor/:supervisorId/history"
+	supervisorSuspendPathPrefix      = "druid/indexer/v1/supervisor/:supervisorId/suspend"
+	supervisorSuspendAllPathPrefix   = "druid/indexer/v1/supervisor/suspendAll"
+	supervisorResumePathPrefix       = "druid/indexer/v1/supervisor/:supervisorId/resume"
+	supervisorResumeAllPathPrefix    = "druid/indexer/v1/supervisor/resumeAll"
+	supervisorResetPathPrefix        = "druid/indexer/v1/supervisor/:supervisorId/reset"
+	supervisorTerminatePathPrefix    = "druid/indexer/v1/supervisor/:supervisorId/terminate"
+	supervisorTerminateAllPathPrefix = "druid/indexer/v1/supervisor/terminateAll"
+	supervisorShutdownPathPrefix     = "druid/indexer/v1/supervisor/:supervisorId/shutdown"
+)
+
+// SupervisorService is a service that submits ingestion tasks to druid supervisor API.
+type SupervisorService struct {
+	client *Client
+}
+
+// CreateSupervisorResponse is a response object of Druid SupervisorService's SubmitTask method
+type CreateSupervisorResponse struct {
+	SupervisorId string `json:"id"`
+}
+
+// TerminateSupervisorResponse is a response object of Druid SupervisorService's Terminate method
+type TerminateSupervisorResponse struct {
+	SupervisorId string `json:"id"`
+}
+
+type GetStatusResponsePayload struct {
+	Datasource      string `json:"dataSource"`
+	Stream          string `json:"stream"`
+	State           string `json:"state"`
+	Partitions      int    `json:"partitions"`
+	Replicas        int    `json:"replicas"`
+	DurationSeconds int    `json:"durationSeconds"`
+}
+
+// GetStatusResponse is a response object of Druid SupervisorService's GetStatus method
+type GetStatusResponse struct {
+	SupervisorId   string                   `json:"id"`
+	GenerationTime time.Time                `json:"generationTime"`
+	Payload        GetStatusResponsePayload `json:"payload"`
+}
+
+// CreateOrUpdate submits an ingestion specification to druid Supervisor API with a pre-configured druid client
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#create-or-update-a-supervisor
+func (s *SupervisorService) CreateOrUpdate(spec interface{}) (string, error) {
```

Comment:
I think it would be better to accept concrete request type instead of `interface{}`

Reply:
Fixed.